src/psycopt2d/load.py

```diff
@@ -0,0 +1,110 @@
+from datetime import date, datetime, timedelta
+from typing import List, Union
+
+import pandas as pd
+from psycopmlutils.loaders import sql_load
+from wasabi import Printer
+
+msg = Printer(timestamp=True)
+
+
+def load_dataset(
+    split_names: Union[List[str], str],
+    drop_if_outcome_before_date: Union[datetime, str],
+    min_lookahead_days: int,
+    datetime_column: str = "pred_timestamp",
+    n_training_samples: Union[None, int] = None,
+) -> pd.DataFrame:
+    """load dataset for t2d.
+
+    Args:
+        split_names (Union[List[str], str]): Names of splits, includes "train", "val",
+            "test".
+        drop_if_outcome_before_date (Union[datetime, str]): Remove patient which have
+            had their outcome before the specified day. Also removed all visits before
+            this date as otherwise the model will learn that all visit before this date
+            does not lead to diabetes.
+            limit the dataset to. Defaults to None.
```

Comment:
```suggestion
        drop_if_outcome_before_date (Union[datetime, str]): Remove patients which 
            experienced an outcome prior to the date. Also removes all visits prior to
            this date as otherwise the model might learn that no visits prior to the date can be tagged with the outcome.
            Defaults to None.
```

`Also removes all visits prior to
            this date as otherwise the model might learn that no visits prior to the date can be tagged with the outcome.`

Hmm, is this a problem? We'd probably not include any events prior to this date in the validation set either, right? :-)